### PR TITLE
allow claimed cluster move to default set

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>
https://github.com/stolostron/backlog/issues/22227#event-6553624161

### CODE REVIEW

File: pkg/webhook/clusterset/validatingWebhook.go [hunk 1]
```diff
@@ -163,6 +164,11 @@ func (a *AdmissionHandler) validateUpdateRequest(request *v1.AdmissionRequest) *
 			return rejectStatus
 		}
 
+		//For upgrade from 2.4 to 2.5, we need to move all clusters from empty set to "default" set.
+		if currentClusterSetName == defaultSetName {
+			return a.allowUpdate(request.UserInfo, originalClusterSetName, currentClusterSetName)
```
Comment (qiujian16) [inline pkg/webhook/clusterset/validatingWebhook.go:169]: I think this should only be bypassed when the cluster is from the claim.  It should be like in line 188

```go
// the cluster is claimed from clusterpool, do not allow to update clustersetlabel.
if clusterDeployment == nil || clusterDeployment.Spec.ClusterPoolRef == nil {
	return a.allowUpdate(request.UserInfo, originalClusterSetName, currentClusterSetName)
}

if len(originalClusterSetName) == 0 && currentClusterSetName == defaultSetName {
    return a.allowUpdate(request.UserInfo, originalClusterSetName, currentClusterSetName)
}

rejectStatus.Result.Message = "Do not allow to update claimed cluster's clusterset."
return rejectStatus

```

File: pkg/webhook/clusterset/validatingWebhook.go [hunk 2]
```diff
@@ -179,6 +180,11 @@ func (a *AdmissionHandler) validateUpdateRequest(request *v1.AdmissionRequest) *
 
 		// the cluster is claimed from clusterpool, do not allow to update clustersetlabel.
 		if clusterDeployment != nil && clusterDeployment.Spec.ClusterPoolRef != nil {
+			//For upgrade from 2.4 to 2.5, we need to move all clusters from empty set to "default" set.
+			if currentClusterSetName == defaultSetName {
```
Comment (zhiweiyin318) [inline pkg/webhook/clusterset/validatingWebhook.go:184]: do we need to check if the old cluster has no clusterset here?

Comment (ldpliu) [reply pkg/webhook/clusterset/validatingWebhook.go]: done

File: pkg/webhook/clusterset/validatingWebhook.go (hunk 1 above)
Comment (ldpliu) [reply pkg/webhook/clusterset/validatingWebhook.go]: done

**Pull request merged**